File: src/types/resource.ts
export type ResourceState =
  | "notInstalled"
  | "installed"
  | {
      downloading: { progress: number; downloaded: number; total: number };
    }
  | { incompatible: { reason: string } }
  | { corrupted: { reason: string } }
  | { failed: { reason: string } };

export interface ResourceCompatibility {
  minMemoryMb: number;
  requiresAvx2: boolean;
  supportedBackends: string[];
  runtimeVersion: string;
}

export interface ResourceItem {
  id: string;
  resourceType: "whisper-model" | "vad-model" | "runtime-asset" | "other";
  name: string;
  version: string;
  sizeBytes: number;
  url: string;
  checksum: string;
  compatibility: ResourceCompatibility;
}

File: src-tauri/src/models/resource.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ResourceType {
    WhisperModel,
    VadModel,
    RuntimeAsset,
    Other,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ResourceItem {
    pub id: String,
    pub resource_type: ResourceType,
    pub name: String,
    pub version: String,
    pub size_bytes: u64,
    pub url: String,
    pub checksum: String,
    pub compatibility: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum ResourceState {
    NotInstalled,
    Downloading {
        progress: f64,
        downloaded: u64,
        total: u64,
    },
    Installed,
    Corrupted,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ResourceManifest {
    pub id: String,
    pub checksum: String,
    pub size_bytes: u64,
    pub installed_at: u64,
}
